### PR TITLE
add pub functionality for zaino

### PR DESCRIPTION
**Pull request by idky137**
## Motivation

For Zaino to use Zebra as its backed it requires access to primitives held in Zebra-Chain and Zebra-RPC.
- This PR will complete Task 1 of https://github.com/zingolabs/zaino/issues/50

## Solution

-Several structs in zebra-rpc were made pub and getter / setter functionality was added.
-From impl was added to ['ConsensusBranchId'] in zebra-chain.

### Tests

The most complex functionality added is simple getter / setter impls, I believe that adding new tests for this would be inefficient / redundant.

### Follow-up Work

Due to patched libraries in Zebra:main causing issues in Zaino these changes were made atop v1.9.0, once the patches are removed we will update the zebra version to 2.0.0, there is a chance that additional changes will be required for changes in zebra v2.0.0.

### PR Author's Checklist

- [ ] The PR name will make sense to users.
- [ ] The PR provides a CHANGELOG summary.
- [ ] The solution is tested.
- [ ] The documentation is up to date.
- [ ] The PR has a priority label.

### PR Reviewer's Checklist

- [ ] The PR Author's checklist is complete.
- [ ] The PR resolves the issue.

### CODE REVIEW

**Comment by idky137** [comment]
I'm not sure whether these changes require a CHANGELOG entry?

File: zebra-rpc/src/methods.rs [hunk 1]
```diff
@@ -1510,11 +1510,23 @@ pub struct AddressBalance {
...
-struct ConsensusBranchIdHex(#[serde(with = "hex")] ConsensusBranchId);
+pub struct ConsensusBranchIdHex(#[serde(with = "hex")] ConsensusBranchId);
+
+impl ConsensusBranchIdHex {
+    /// Returns a new instance of ['ConsensusBranchIdHex'].
+    pub fn new(consensus_branch_id: u32) -> Self {
+        ConsensusBranchIdHex(consensus_branch_id.into())
+    }
+
+    /// Returns a reference to the ['ConsensusBranchId'].
```
Comment (oxarbitrage) [inline zebra-rpc/src/methods.rs]:
```suggestion
    /// Returns the value of the ['ConsensusBranchId'].
```

**Review by oxarbitrage — DISMISSED** [review]
Looks good to me, i left a minor suggestion for a doc change.

@gustavovalverde is working on a CI issue we have with external pull requests so we can merge this one. 

Thanks!

File: zebra-rpc/src/methods.rs (hunk 1 above)
Comment (idky137) [reply zebra-rpc/src/methods.rs]: Brilliant! I have added the suggested change in [308c9c6].

**Comment by idky137** [comment]
Please let me know if there is anything I can do to help remove the 'Do-Not-Merge' Tag..

**Comment by mpguerra** [comment]
> Please let me know if there is anything I can do to help remove the 'Do-Not-Merge' Tag..

It was just while we were prepping the release. We should be good to merge now that's out :)

**Comment by oxarbitrage** [comment]
Hi @idky137 can you update your branch with the last `main` ? We [just merged some fixes to the external contributor workflow](https://github.com/ZcashFoundation/zebra/pull/8988) and we want to check if it is working.

**Review by oxarbitrage — APPROVED** [review]

**Pull request merged**